### PR TITLE
Added Merge Sort [Golang]

### LINKED ISSUE #356: Merge Sort [Golang]
#### This issue is about [Merge Sort](https://www.khanacademy.org/computing/computer-science/algorithms/merge-sort/a/overview-of-merge-sort).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by rashinanwani**
Fixes #356

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by aashutoshrathi** [comment]
Why does it have `euclidean_gcd/euclidean_gcd.go` rebase please.

File: merge_sort/merge_sort.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package main
```
Comment (Prakash2403) [inline merge_sort/merge_sort.go]: Mention space and time complexity for each function.

File: merge_sort/merge_sort.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+// Merge merges two sorted slice
+// Time complexity : O(size)
+// Space complexity : O(size)
+func Merge(left []int, right []int) []int {
+	i := 0
+	j := 0
+	size := len(left) + len(right)
+	slice := make([]int, size, size) // make built-in function allocates and initializes an object of type slice
+	for k := 0; k < size; k++ {
+		if i > len(left)-1 && j <= len(right)-1 {
```
Comment (Monal5031) [inline merge_sort/merge_sort.go]: `j < len(right)` would look good here.

File: merge_sort/merge_sort.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+			slice[k] = right[j]
+			j++
+		} else if j > len(right)-1 && i <= len(left)-1 {
```
Comment (Monal5031) [inline merge_sort/merge_sort.go]: `i < len(left)` would look good here.

File: merge_sort/merge_sort.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+// merge merges two sorted slice
+// Time complexity : O(size)
+// Space complexity : O(size)
+func merge(left []int, right []int) []int {
+	i := 0
+	j := 0
+	size := len(left) + len(right)
+	slice := make([]int, size, size) // make built-in function allocates and initializes an object of type slice
```
Comment (mohitkyadav) [inline merge_sort/merge_sort.go]: Line too long.

**Pull request merged**